website/developer-docs: add notice about libxmlsec compatibility (#5553)
goauthentik/authentik: website/developer-docs/setup/full-dev-environment.md

## Backend Setup

:::info

> Depending on platform, some native dependencies might be required. On macOS, run `brew install libxmlsec1 libpq`, and for the CLI tools `brew install postgresql redis node@20`

Reviewer: ```suggestion
Depending on your platform, some native dependencies might be required. On MacOS, run `brew install libxmlsec1 libpq`, and for the CLI tools `brew install postgresql redis node@20`
```

Author: macOS is actually stylized lowercase nowadays https://en.wikipedia.org/wiki/MacOS

```suggestion
Depending on your platform, some native dependencies might be required. On macOS, run `brew install libxmlsec1 libpq`, and for the CLI tools `brew install postgresql redis node@20`
```

Reviewer: Eventually we need to decide either how to handle writing with OS-agnosticism, or... provide a tabbed .mdx section with commands for each OS.

:::

:::info

> While [this issue](https://github.com/xmlsec/python-xmlsec/issues/252) is open, a workaround is required to install a compatible version of `libxmlsec1` with brew, see [this comment](https://github.com/xmlsec/python-xmlsec/issues/254#issuecomment-1511135314).

Reviewer: ```suggestion
As long as [this issue](https://github.com/xmlsec/python-xmlsec/issues/252) is open, a workaround is required to install a compatible version of `libxmlsec1` with brew, see [this comment](https://github.com/xmlsec/python-xmlsec/issues/254#issuecomment-1511135314).
```
